diff -rupN old_binutils-2.28/bfd/config.bfd new_binutils-2.28/bfd/config.bfd
--- old_binutils-2.28/bfd/config.bfd	2017-03-02 08:23:53.000000000 +0000
+++ new_binutils-2.28/bfd/config.bfd	2017-09-30 13:53:48.674427702 +0100
@@ -348,7 +348,6 @@ case "${targ}" in
   am33_2.0-*-linux*)
     targ_defvec=am33_elf32_linux_vec
     ;;
-
   arc*eb-*-elf* | arc*eb-*-linux*)
     targ_defvec=arc_elf32_be_vec
     targ_selvecs=arc_elf32_le_vec
@@ -358,7 +357,18 @@ case "${targ}" in
     targ_defvec=arc_elf32_le_vec
     targ_selvecs=arc_elf32_be_vec
     ;;
-
+  i[3-7]86-*-dbos*)
+    targ_defvec=i386_elf32_vec
+    targ_selvec=
+    targ64_selvecs=x86_64_elf64_vec
+    ;;
+#ifdef BFD64
+  x86_64-*-dbos*)
+    targ_defvec=x86_64_elf64_vec
+    targ_selvecs=i386_elf32_vec
+    want64=true
+    ;;
+#endif
   arm-*-darwin*)
     targ_defvec=arm_mach_o_vec
     targ_selvecs="mach_o_le_vec mach_o_be_vec mach_o_fat_vec"
diff -rupN old_binutils-2.28/config.sub new_binutils-2.28/config.sub
--- old_binutils-2.28/config.sub	2016-12-23 08:40:17.000000000 +0000
+++ new_binutils-2.28/config.sub	2017-09-30 13:48:43.290967585 +0100
@@ -1384,7 +1384,7 @@ case $os in
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
-	      | -sym* | -kopensolaris* | -plan9* \
+	      | -sym* | -kopensolaris* | -plan9* | -dbos* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
 	      | -aos* | -aros* | -cloudabi* | -sortix* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
diff -rupN old_binutils-2.28/gas/configure.tgt new_binutils-2.28/gas/configure.tgt
--- old_binutils-2.28/gas/configure.tgt	2017-03-02 08:23:53.000000000 +0000
+++ new_binutils-2.28/gas/configure.tgt	2017-09-30 13:55:52.302210576 +0100
@@ -6,12 +6,12 @@
 # it under the terms of the GNU General Public License as published by
 # the Free Software Foundation; either version 3 of the License, or
 # (at your option) any later version.
-# 
+#
 # This program is distributed in the hope that it will be useful,
 # but WITHOUT ANY WARRANTY; without even the implied warranty of
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 # GNU General Public License for more details.
-# 
+#
 # You should have received a copy of the GNU General Public License
 # along with this program; see the file COPYING3.  If not see
 # <http://www.gnu.org/licenses/>.
@@ -220,6 +220,7 @@ case ${generic_target} in
   i386-sequent-bsd*)			fmt=aout em=dynix ;;
   i386-*-beospe*)			fmt=coff em=pe ;;
   i386-*-beos*)				fmt=elf ;;
+  i386-*-dbos*)       fmt=elf em=gnu ;;
   i386-*-coff)				fmt=coff ;;
   i386-*-elfiamcu)			fmt=elf arch=iamcu ;;
   i386-*-elf*)				fmt=elf ;;
@@ -359,7 +360,7 @@ case ${generic_target} in
   moxie-*-uclinux)			fmt=elf em=linux ;;
   moxie-*-moxiebox*)                    fmt=elf endian=little ;;
   moxie-*-*)				fmt=elf ;;
-    
+
   mt-*-elf)				fmt=elf bfd_gas=yes ;;
 
   msp430-*-*)				fmt=elf ;;
@@ -470,7 +471,7 @@ case ${generic_target} in
   visium-*-elf)				fmt=elf ;;
 
   xstormy16-*-*)			fmt=elf ;;
-  
+
   xgate-*-*)    			fmt=elf ;;
 
   xtensa*-*-*)				fmt=elf ;;
diff -rupN old_binutils-2.28/ld/configure.tgt new_binutils-2.28/ld/configure.tgt
--- old_binutils-2.28/ld/configure.tgt	2017-03-02 08:23:54.000000000 +0000
+++ new_binutils-2.28/ld/configure.tgt	2017-09-30 14:03:18.526960710 +0100
@@ -255,6 +255,15 @@ i[3-7]86-*-linux*aout*)	targ_emul=i386li
 			targ_extra_emuls="elf_i386 elf_iamcu"
 			tdir_elf_iamcu=`echo ${targ_alias} | sed -e 's/aout//'`
 			tdir_elf_i386=`echo ${targ_alias} | sed -e 's/aout//'` ;;
+i[3-7]86-*-dbos*)
+			targ_emul=elf_i386_dbos
+			targ_extra_emuls=elf_i386
+			targ64_extra_emuls="elf_x86_64_dbos elf_x86_64"
+			;;
+x86_64-*-dbos*)
+			targ_emul=elf_x86_64_dbos
+			targ_extra_emuls="elf_i386_dbos elf_x86_64 elf_i386"
+			;;
 i[3-7]86-*-linux*oldld)	targ_emul=i386linux
 			targ_extra_emuls="elf_i386 elf_iamcu" ;;
 i[3-7]86-*-linux-*)	targ_emul=elf_i386
diff -rupN old_binutils-2.28/ld/emulparams/elf_i386_dbos.sh new_binutils-2.28/ld/emulparams/elf_i386_dbos.sh
--- old_binutils-2.28/ld/emulparams/elf_i386_dbos.sh	1970-01-01 01:00:00.000000000 +0100
+++ new_binutils-2.28/ld/emulparams/elf_i386_dbos.sh	2017-09-30 14:03:50.183266456 +0100
@@ -0,0 +1,15 @@
+SCRIPT_NAME=elf
+OUTPUT_FORMAT="elf32-i386"
+CHECK_RELOCS_AFTER_OPEN_INPUT=yes
+NO_RELA_RELOCS=yes
+TEXT_START_ADDR=0x10000000
+MAXPAGESIZE="CONSTANT (MAXPAGESIZE)"
+COMMONPAGESIZE="CONSTANT (COMMONPAGESIZE)"
+ARCH=i386
+MACHINE=
+TEMPLATE_NAME=elf32
+GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
+NO_SMALL_DATA=yes
+SEPARATE_GOTPLT="SIZEOF (.got.plt) >= 12 ? 12 : 0"
+IREL_IN_PLT=
diff -rupN old_binutils-2.28/ld/emulparams/elf_x86_64_dbos.sh new_binutils-2.28/ld/emulparams/elf_x86_64_dbos.sh
--- old_binutils-2.28/ld/emulparams/elf_x86_64_dbos.sh	1970-01-01 01:00:00.000000000 +0100
+++ new_binutils-2.28/ld/emulparams/elf_x86_64_dbos.sh	2017-09-30 14:06:10.720429461 +0100
@@ -0,0 +1 @@
+. ${srcdir}/emulparams/elf_x86_64.sh
diff -rupN old_binutils-2.28/ld/Makefile.am new_binutils-2.28/ld/Makefile.am
--- old_binutils-2.28/ld/Makefile.am	2017-03-02 08:23:54.000000000 +0000
+++ new_binutils-2.28/ld/Makefile.am	2017-09-30 14:13:07.422577613 +0100
@@ -296,6 +296,7 @@ ALL_EMULATION_SOURCES = \
 	eelf_i386.c \
 	eelf_i386_be.c \
 	eelf_i386_chaos.c \
+	eelf_i386_dbos.c \
 	eelf_i386_fbsd.c \
 	eelf_i386_ldso.c \
 	eelf_i386_nacl.c \
@@ -511,6 +512,7 @@ ALL_64_EMULATION_SOURCES = \
 	eelf_k1om.c \
 	eelf_k1om_fbsd.c \
 	eelf_x86_64.c \
+	eelf_x86_64_dbos.c \
 	eelf_x86_64_cloudabi.c \
 	eelf_x86_64_fbsd.c \
 	eelf_x86_64_nacl.c \
@@ -1333,6 +1335,10 @@ eelf32xtensa.c: $(srcdir)/emulparams/elf
 eelf_i386.c: $(srcdir)/emulparams/elf_i386.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_i386_dbos.c: $(srcdir)/emulparams/elf_i386_dbos.sh \
+	$(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+		${GENSCRIPTS} elf_i386_dbos "$(tdir_elf_i386_dbos)"
+
 eelf_i386_be.c: $(srcdir)/emulparams/elf_i386_be.sh \
   $(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
@@ -2032,6 +2038,10 @@ eelf_k1om_fbsd.c: $(srcdir)/emulparams/e
 eelf_x86_64.c: $(srcdir)/emulparams/elf_x86_64.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
 
+eelf_x86_64_dbos.c: $(srcdir)/emulparams/elf_x86_64_dbos.sh \
+	$(ELF_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
+		${GENSCRIPTS} elf_x86_64_dbos "$(tdir_elf_x86_64_dbos)"
+
 eelf_x86_64_cloudabi.c: $(srcdir)/emulparams/elf_x86_64_cloudabi.sh \
   $(srcdir)/emulparams/elf_x86_64.sh \
   $(ELF_X86_DEPS) $(srcdir)/scripttempl/elf.sc ${GEN_DEPENDS}
