diff -rupN host_old_gcc-7.1.0/config.sub host_gcc-7.1.0/config.sub
--- host_old_gcc-7.1.0/config.sub	2016-12-29 11:04:53.000000000 +0000
+++ host_gcc-7.1.0/config.sub	2017-09-30 02:47:46.911098105 +0100
@@ -1409,7 +1409,7 @@ case $os in
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
-	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox*)
+	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -dbos*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN host_old_gcc-7.1.0/gcc/config/dbos.h host_gcc-7.1.0/gcc/config/dbos.h
--- host_old_gcc-7.1.0/gcc/config/dbos.h	1970-01-01 01:00:00.000000000 +0100
+++ host_gcc-7.1.0/gcc/config/dbos.h	2017-09-30 02:51:31.923224209 +0100
@@ -0,0 +1,18 @@
+#undef TARGET_DBOS
+#define TARGET_DBOS 1
+#define LIB_SPEC "%{pthread:-lpthread} -lc"
+#undef STARTFILE_SPEC
+#define STARTFILE_SPEC "%{!shared: %{!pg:crt0.o%s}} crti.o%s %{!shared:crtbegin.o%s}"
+#undef ENDFILE_SPEC
+#define ENDFILE_SPEC "%{!shared:crtend.o%s} crtn.o%s"
+#undef  NO_IMPLICIT_EXTERN_C
+#define NO_IMPLICIT_EXTERN_C 1
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+  do {                                \
+    builtin_define ("__dbos__");      \
+    builtin_define ("__unix__");      \
+    builtin_assert ("system=dbos");   \
+    builtin_assert ("system=unix");   \
+    builtin_assert ("system=posix");   \
+  } while(0);
diff -rupN host_old_gcc-7.1.0/gcc/config.gcc host_gcc-7.1.0/gcc/config.gcc
--- host_old_gcc-7.1.0/gcc/config.gcc	2017-03-24 13:59:51.000000000 +0000
+++ host_gcc-7.1.0/gcc/config.gcc	2017-09-30 02:50:07.123175880 +0100
@@ -174,7 +174,7 @@
 #			the --with-sysroot configure option or the
 #			--sysroot command line option is used this
 #			will be relative to the sysroot.
-# target_type_format_char 
+# target_type_format_char
 # 			The default character to be used for formatting
 #			the attribute in a
 #			.type symbol_name, ${t_t_f_c}<property>
@@ -665,6 +665,11 @@ case ${target} in
   default_use_cxa_atexit=yes
   use_gcc_stdint=wrap
   ;;
+*-*-dbos*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=yes
+  ;;
 *-*-freebsd*)
   # This is the generic ELF configuration of FreeBSD.  Later
   # machine-specific sections may refine and add to this
@@ -918,6 +923,12 @@ case ${target} in
 esac
 
 case ${target} in
+i[34567]86-*-dbos*)
+    tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h dbos.h"
+    ;;
+x86_64-*-dbos*)
+    tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h glibc-stdint.h i386/i386elf.h i386/x86-64.h dbos.h"
+    ;;
 aarch64*-*-elf | aarch64*-*-fuchsia* | aarch64*-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h newlib-stdint.h"
 	tm_file="${tm_file} aarch64/aarch64-elf.h aarch64/aarch64-elf-raw.h"
@@ -3384,7 +3395,7 @@ if test x$with_cpu = x ; then
          with_cpu=8548
       else
          with_cpu=8540
-      fi       
+      fi
       ;;
     sparc*-*-*)
       case ${target} in
diff -rupN host_old_gcc-7.1.0/libstdc++-v3/crossconfig.m4 host_gcc-7.1.0/libstdc++-v3/crossconfig.m4
--- host_old_gcc-7.1.0/libstdc++-v3/crossconfig.m4	2017-01-17 23:36:12.000000000 +0000
+++ host_gcc-7.1.0/libstdc++-v3/crossconfig.m4	2017-09-30 02:52:48.411268486 +0100
@@ -93,7 +93,7 @@ case "${host}" in
 
   *-freebsd*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
-    AC_SUBST(SECTION_FLAGS) 
+    AC_SUBST(SECTION_FLAGS)
     GLIBCXX_CHECK_LINKER_FEATURES
     AC_DEFINE(HAVE_SETENV)
     AC_DEFINE(HAVE_FINITEF)
@@ -140,6 +140,13 @@ case "${host}" in
     AC_SUBST(SECTION_FLAGS)
     ;;
 
+  *-dbos*)
+    GLIBCXX_CHECK_COMPILER_FEATURES
+    GLIBCXX_CHECK_LINKER_FEATURES
+    GLIBCXX_CHECK_MATH_SUPPORT
+    GLIBCXX_CHECK_STDLIB_SUPPORT
+    ;;
+
   *-hpux*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
     AC_SUBST(SECTION_FLAGS)
@@ -200,7 +207,7 @@ case "${host}" in
     ;;
   *-netbsd*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
-    AC_SUBST(SECTION_FLAGS) 
+    AC_SUBST(SECTION_FLAGS)
     GLIBCXX_CHECK_LINKER_FEATURES
     AC_DEFINE(HAVE_FINITEF)
     AC_DEFINE(HAVE_FINITE)
@@ -218,7 +225,7 @@ case "${host}" in
     ;;
   *-qnx6.1* | *-qnx6.2*)
     SECTION_FLAGS='-ffunction-sections -fdata-sections'
-    AC_SUBST(SECTION_FLAGS) 
+    AC_SUBST(SECTION_FLAGS)
     GLIBCXX_CHECK_LINKER_FEATURES
     AC_DEFINE(HAVE_COSF)
     AC_DEFINE(HAVE_COSL)
@@ -245,8 +252,8 @@ case "${host}" in
     AC_DEFINE(HAVE_FINITE)
     AC_DEFINE(HAVE_FPCLASS)
     # All of the dependencies for wide character support are here, so
-    # turn it on. 
-    AC_DEFINE(_GLIBCXX_USE_WCHAR_T) 
+    # turn it on.
+    AC_DEFINE(_GLIBCXX_USE_WCHAR_T)
     # These two C99 functions are present only in Solaris >= 10
     AC_DEFINE(HAVE_STRTOF)
     AC_DEFINE(HAVE_STRTOLD)
